File: back/coursDevOps/src/main/java/edu/esiea/coursDevOps/models/Product.java
package edu.esiea.coursDevOps.models;

import java.util.Objects;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Product {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String name;
	private String description;
	private String imgUrl;
	private float basePrice;
	private float tva;
	
	public Product() {}

	public Product(int id, String name, String description, String imgUrl, float basePrice, float tva) {
		super();
		this.id = id;
		this.name = name;
		this.description = description;
		this.imgUrl = imgUrl;
		this.basePrice = basePrice;
		this.tva = tva;
	}


	public int getId() {
		return id;
	}


	public void setId(int id) {
		this.id = id;
	}


	public String getName() {
		return name;
	}


	public void setName(String name) {
		this.name = name;
	}


	public String getDescription() {
		return description;
	}


	public void setDescription(String description) {
		this.description = description;
	}


	public String getImgUrl() {
		return imgUrl;
	}


	public void setImgUrl(String imgUrl) {
		this.imgUrl = imgUrl;
	}


	public float getBasePrice() {
		return basePrice;
	}


	public void setBasePrice(float basePrice) {
		this.basePrice = basePrice;
	}


	public float getTva() {
		return tva;
	}


	public void setTva(float tva) {
		this.tva = tva;
	}
	
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Product product = (Product) o;
        return id == product.id &&
                Float.compare(product.basePrice, basePrice) == 0 &&
                Float.compare(product.tva, tva) == 0 &&
                Objects.equals(name, product.name) &&
                Objects.equals(description, product.description) &&
                Objects.equals(imgUrl, product.imgUrl);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, description, imgUrl, basePrice, tva);
    }
}

File: front/coursDevOps/src/app/models/product.ts
export class Product {
    private _basePrice!: number;
    private _tva!: number;
    private _name!: string;
    private _imgUrl!: string;

    constructor(basePrice: number, tva: number, name: string, imgUrl: string){
        this._basePrice = basePrice;
        this._tva = tva;
        this._name = name;
        this._imgUrl = imgUrl;
    }

    public get basePrice(): number {
        return this._basePrice;
    }
    public set basePrice(value: number) {
        this._basePrice = value;
    }

    public get tva(): number {
        return this._tva;
    }
    public set tva(value: number) {
        this._tva = value;
    }

    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }

    public get imgUrl(): string {
        return this._imgUrl;
    }
    public set imgUrl(value: string) {
        this._imgUrl = value;
    }
}
